tests/Linter/ProcessLintingResultTest.php

```diff
@@ -24,6 +24,17 @@
  */
 final class ProcessLintingResultTest extends TestCase
 {
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+
+        /** @see https://github.com/phpspec/prophecy/issues/527 */
+        $type = (new \ReflectionMethod(\Symfony\Component\Process\Process::class, 'mustRun'))->getReturnType();
+        if ($type instanceof \ReflectionNamedType && 'static' === $type->getName()) {
+            static::markTestSkipped('Prophecy cannot handle the static return type.');
```

Comment:
The test is green if I rewrite it to PHPUnit mocks (and disable the check that disallows me to do that). I don't know if that's enough for you. Otherwise somebody has to fix Prophecy. 😇

Reply:
unfortunately current manual check you did doesn't prevent an accidental break in future, when nobody would be manually checking again.
Not sure how fast Prophecy can propose a fix?

happy for more voices in discussion ;)

Reply:
I tend to agree with @keradus. PHP CS Fixer itself does not actually need Symfony 6.x, using 4.x/5.x is just fine. Allowing 6.x is mostly needed for projects that include PHP CS Fixer in their own dependencies, which is wrong IMO—it should be installed separately to avoid conflicts. While allowing 6.x would help those projects, I don't think it's worth the trade-off of losing this test. I would wait for Prophecy to be fixed.

Reply:
To put this into perspective: You wouldn't lose the test as it is still executed for Symfony 4&5. This PR would create a pretty small and temporary blind spot for a Symfony release that is still in beta. The risk is pretty low, but I also understand that we're not in a hurry.

Reply:
> This PR would create a pretty small and temporary blind spot for a Symfony release that is still in beta.

- it's not small. if we are not sure if linting is working, we can break things for end users
- is it's temporary, I'm OK to wait extra week or 2 and have v6 support only afterwards, if it's not temporary - we need another solution than skipping the tests

Reply:
> we need another solution than skipping the tests

Our options are:

* Patch Prophecy
* Find a way to run this test without mocking the process object
* Use a different mock solution

Reply:
> if we are not sure if linting is working

We can be sure it's working because this functionality should be covered by `PhpCsFixer\Tests\Linter\ProcessLinterTest` already. The test I'm skipping here conditionally is pretty much redundant.